src/compiler/checker.ts

```diff
@@ -40095,10 +40095,10 @@ namespace ts {
                         }
                         // ECMA262: 15.2.1.1 It is a Syntax Error if the ExportedNames of ModuleItemList contains any duplicate entries.
                         // (TS Exceptions: namespaces, function overloads, enums, and interfaces)

```

Comment:
it doesn't look like exported function overloads are handled here. can you add a test case and update the comment if needed?

Reply:
added a couple tests for this and it seems to be working correctly. i believe it's handled [here](https://github.com/microsoft/TypeScript/pull/46871/files#diff-d9ab6589e714c71e657f601cf30ff51dfc607fc98419bf72e04f6b0fa92cc4b8R40101)